ee/secureenclavesigner/secureenclavesigner.go

```diff
@@ -0,0 +1,234 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"strings"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/kolide/launcher/ee/allowedcmd"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const (
+	CreateKeyCmd = "create-key"
+	SignCmd      = "sign"
+)
+
+type SignRequest struct {
+	Request
+	// Digest is the hash []byte of the data to be signed
+	Digest []byte `msgpack:"digest"`
+	// SecureEnclavePubKey is the B64 encoded DER of the public key to be used to verify the signature
+	SecureEnclavePubKey []byte `msgpack:"secure_enclave_pub_key"`
+}
+
+type Request struct {
```

Comment:
I'm not sure I understand `SignRequest` and `Request`. 

`Request` feels too generic. And I'm not sure the Digest in SignRequest is protected by the signature Can you walk me through them?

Reply:
We talked about this offline some, but I've added a CreateKeyReqeust object to try to help clarify.